subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/ConventionReportTask.java

```diff
@@ -70,7 +69,7 @@ public DirectoryProperty getProjectReportDirectory() {
 
     protected ConventionReportTask() {
         reportDir = getProject().getObjects().directoryProperty();
-        getOutputs().upToDateWhen(satisfyNone());
+        doNotTrackState("Uses the whole project state as an input");
```

Comment:
We do it like this because the task is going to be extended by others? I wonder if it would make sense to start inheriting these annotations, maybe in 8.0...

Reply:
Yeah, that is exactly the case. I think it would make sense to inherit the disabling annotations and not inherit the enabling annotations. So inherit `@UntrackedTask` and don't inherit `@CacheableTask`. Though I wonder how much better this is afterwards.